include/oneapi/dpl/pstl/hetero/dpcpp/parallel_backend_sycl.h

```diff
@@ -879,13 +879,45 @@ __parallel_transform_scan(oneapi::dpl::__internal::__device_backend_tag __backen
         }
     }
 
-    oneapi::dpl::__par_backend_hetero::__gen_transform_input<_UnaryOperation>
-        __gen_transform{__unary_op};
-    return __future(__parallel_transform_reduce_then_scan(
-                        __backend_tag, ::std::forward<_ExecutionPolicy>(__exec), ::std::forward<_Range1>(__in_rng),
-                        ::std::forward<_Range2>(__out_rng), __gen_transform, __binary_op, __gen_transform,
-                        oneapi::dpl::__internal::__no_op{}, __simple_write_to_idx{}, __init, _Inclusive{})
-                        .event());
+    // Reduce-then-scan is dependent on sycl::shift_group_right which requires the underlying type to be trivially
+    // copyable. If this is not met, then we must fallback to the legacy implementation.
+    if constexpr (std::is_trivially_copyable_v<_Type>)
+    {
+        oneapi::dpl::__par_backend_hetero::__gen_transform_input<_UnaryOperation> __gen_transform{__unary_op};
+        return __future(__parallel_transform_reduce_then_scan(
+                            __backend_tag, std::forward<_ExecutionPolicy>(__exec), std::forward<_Range1>(__in_rng),
+                            std::forward<_Range2>(__out_rng), __gen_transform, __binary_op, __gen_transform,
+                            oneapi::dpl::__internal::__no_op{}, __simple_write_to_idx{}, __init, _Inclusive{})
+                            .event());
+    }
+    else
+    {
+        // Either we can't use group scan or this input is too big for one workgroup
```

Comment:
I don't think this comment is correct at this point, since we are now falling back if we can't use group scan, or use `shift_group_right`, or the input is larger than a single work-group. Maybe we can just remove the comment.

Reply:
I agree we do not need this comment and have removed it.